packages/chakra-components/src/MultiSelect/index.js

```diff
@@ -0,0 +1,756 @@
+      >
+        {children({ selectedOption: selectedOptions[0] })}
+      </PseudoBox>
+    );
+  }
+
+  return null;
+};
+
+MultiSelectSelectedOption.displayName = "MultiSelectSelectedOption";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const MultiSelectPlaceholder = ({ children, ...props }) => {
+  const { values, inputValue } = useMultiSelectContext();
+
+  const theme = useTheme();
+  const { colorMode } = useColorMode();
+
+  const placeholderColor = {
+    light: theme.colors.gray[400],
+    dark: theme.colors.whiteAlpha[400],
+  };
+
+  if (!values.length) {
+    return inputValue ? null : (
+      <PseudoBox
+        position="absolute"
+        top="50%"
+        transform="translateY(-50%)"
+        mx="2px"
+        color={placeholderColor[colorMode]}
+        {...props}
+      >
+        {children}
+      </PseudoBox>
+    );
+  }
+
+  return null;
+};
+
+MultiSelectPlaceholder.displayName = "MultiSelectPlaceholder";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const MultiSelectInput = forwardRef(
+  (
+    {
+      "aria-label": ariaLabel,
+      "aria-labelledby": ariaLabelledby,
+      isReadOnly,
+      isDisabled,
+      isInvalid,
+      isRequired,
+      ...props
+    },
+    ref,
+  ) => {
+    const { inputIsHidden, getInputProps, isListBox } = useMultiSelectContext();
+
+    const ariaAttributes = {
+      "aria-autocomplete": "list",
+      "aria-label": ariaLabel,
+      "aria-labelledby": ariaLabelledby,
+    };
+
+    const { inputWrapperStyle, inputStyle } = useMultiSelectInputStyle({
+      isDisabled,
+      inputIsHidden,
+      isListBox,
+    });
+
+    return (
+      <PseudoBox {...inputWrapperStyle} {...props}>
+        <AutosizeInput
+          type="text"
+          autoCapitalize="none"
+          autoComplete="off"
+          autoCorrect="off"
+          spellCheck="false"
+          inputStyle={inputStyle}
+          disabled={isDisabled}
+          {...ariaAttributes}
+          {...getInputProps()}
+        />
+      </PseudoBox>
+    );
+  },
+);
+
+MultiSelectInput.displayName = "MultiSelectInput";
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const MultiSelectInputGroup = props => {
+  const {
+    isMulti,
+    placeholder,
+    renderCustomPlaceholder,
+    renderCustomSelectedOption,
+    renderCustomInput,
+  } = useMultiSelectContext();
+
+  const renderSelectedOption = () => {
+    if (isMulti) {
+      return (
+        <MultiSelectSelectedOption>
+          {({ selectedOption, handleOnClick }) => (
+            <Tag size="md" variant="solid" variantColor="blue">
+              <TagLabel>{selectedOption.label}</TagLabel>
+              <TagCloseButton tabIndex={-1} onClick={handleOnClick} />
+            </Tag>
+          )}
+        </MultiSelectSelectedOption>
```

Comment:
Can I Customize this tag from outside ?

Reply:
Style wise, haven't worked on that yet. Chakra's variants including.
Planning to improve this on the next iteration.
Your thoughts?

Reply:
Yeah that is fine. We might get a better idea when we comeback to this.